apps/sparo-lib/src/services/GitService.ts

```diff
@@ -438,6 +438,66 @@ Please specify a directory on the command line
     return objectType;
   }
 
+  public getCurrentBranch(): string {
+    const currentBranch: string = this.executeGitCommandAndCaptureOutput({
+      args: ['branch', '--show-current']
+    }).trim();
+    return currentBranch;
+  }
+
+  /**
+   * Check existence for a list of branch name
+   */
+  public checkRemoteBranchesExistenceAsync = async (
+    remote: string,
+    branches: string[]
+  ): Promise<Record<string, boolean>> => {
+    this._terminalService.terminal.writeDebugLine(`Checking branches: ${branches.join(',')}`);
+    const ret: Record<string, boolean> = {};
+    await Async.forEachAsync(branches, async (branch: string) => {
+      const isExists: boolean = await this.checkRemoteBranchExistenceAsync(remote, branch);
+      ret[branch] = isExists;
+    });
+    return ret;
+  };
+
+  /**
+   * Check existence for one branch name.
+   *
+   * Function "checkRemoteBranchesExistenceAsync" is preferred if you are going to check a list of branch name.
+   */
+  public checkRemoteBranchExistenceAsync = async (remote: string, branch: string): Promise<boolean> => {
+    const gitPath: string = this.getGitPathOrThrow();
+    const currentWorkingDirectory: string = this.getRepoInfo().root;
+    const childProcess: child_process.ChildProcess = Executable.spawn(
+      gitPath,
+      ['ls-remote', '--exit-code', remote, branch],
+      {
+        currentWorkingDirectory,
+        stdio: ['ignore', 'pipe', 'pipe']
+      }
+    );
+    if (!childProcess.stdout || !childProcess.stderr) {
+      this._terminalService.terminal.writeDebugLine(`Failed to spawn git process, fallback to spawnSync`);
+      const result: string = this.executeGitCommandAndCaptureOutput({
+        args: ['ls-remote', remote, branch]
+      }).trim();
+      return Promise.resolve(!!result);
+    }
+    return await new Promise((resolve, reject) => {
+      // Only care about exit code since specifying --exit-code
+      childProcess.on('close', (exitCode: number | null) => {
```

Comment:
We just did a big PR for Rush Stack to correctly handle `signal` as well as `exitCode`:

https://github.com/microsoft/rushstack/pull/4721

Reply:
If the subprocess is terminated by a signal, then `if (exitCode)` will be `null` which is falsy, and your `(`Branch "${branch}" exists remotely`);` will get printed incorrectly.

The fix is easy --- just copy what that PR does